packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -90,10 +90,26 @@
 #define IFPACK2_BLOCKTRIDICONTAINER_USE_MPI_3
 #endif
 
+// this requires kokkos develop branch (do not use this yet)
+//#define IFPACK2_BLOCKTRIDICONTAINER_USE_CUDA_STREAM
+
+// if defined, this use pinned memory instead of device pointer, cuda aware mpi
+#define IFPACK2_BLOCKTRIDICONTAINER_USE_PINNED_MEMORY_FOR_MPI
+
+// if defined, all views are allocated on cuda space intead of cuda uvm space
+// #define IFPACK2_BLOCKTRIDICONTAINER_USE_CUDA_SPACE
+
 namespace Ifpack2 {
 
   namespace BlockTriDiContainerDetails {
 
+    /// for the next promotion
```

Comment:
I do wish kokkos-kernels had promotion number macros :) .  Nothing we can do about that now, though.

Reply:
I just put the temporary macro in kokkoskernels when I need to include changes related to the promotion. After promotion is done, I remove the macro from kokkoskernels and also remove it from ifpack2. It is not systematic but works ; )

Reply:
No worries :)